### PR TITLE
Pipeline config/executable

### PR DESCRIPTION
**Pull request by malchua**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR allows an administrator of IRIDA Next to specify which versions of pipelines are not executable.
This PR relates to issue #680.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Sign in
2. Select Projects on the sidebar and click on a project that is listed
3. Select Samples in the side bar and select one or two of the samples
4. Click the rocket-ship button to start creating a workflow execution
5. Compare the pipelines that are listed with the pipelines listed in `config/pipelines/pipelines.json`
6. Check to see that only the pipelines that do not have the executable option set to `false` are listed
7. Select Settings in the side bar and select Automated Workflow Executions from the drop-down menu
8. Click on New automated workflow execution button
9. Compare the pipelines that are listed with the pipelines listed in `config/pipelines/pipelines.json`
10. Check to see that only the pipelines with `automatable: true' and not `executable: false` are listed

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by JeffreyThiessen — APPROVED** [review]
Look great. Tested and working as intended. :+1:

**Review by ChrisHuynh333 — CHANGES_REQUESTED** [review]
If I have an automated workflow execution set up (when `executable` was `true`, and then its selected pipeline's `executable` is set to `false`, an error occurs if I go to edit the automated workflow execution.

**Review by ChrisHuynh333 — CHANGES_REQUESTED** [review]
Last one from me.
If a workflow execution is started, and the selected pipeline's `executable` is set to `false` while the workflow execution is in state `new/initial`, it will hang and stay in this state because the pipeline is no longer able to be found. In talking to Eric, the state should be updated to `error` in these cases.

You will want to add error handling to the `preparation_service` like we have for other services and add an appropriate message in case in the future, we want to display the error msg back to the user. You will also want to set `.cleaned = true` in order to for the WE to be deletable. We do not need to actually go through the `cleanup_service` as there is nothing that requires cleaning

**Comment by ericenns** [comment]
> I changed it so that even if the automated workflow execution does not have an executable pipeline, they can still see it and can edit it. Should I instead change that to so they won't be able to see it if it's not executable?

I was thinking that we need some way to inform the user that the `automated_workflow_execution` will not work since it  is either no longer executable or automatable. For now I think your solution is sufficient, but I think it should be addressed in a follow up PR.

File: app/services/workflow_executions/preparation_service.rb [hunk 1]
```diff
@@ -11,12 +11,22 @@ class PreparationService < BaseService
...
 
-    def execute # rubocop:disable Metrics/MethodLength
+    def execute # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+      # confirm pipeline found
+      if @pipeline.nil?
+        @workflow_execution.state = :error
+        @workflow_execution.cleaned = true
+        @workflow_execution.save
+        return false
+      end
```
Comment (ChrisHuynh333) [inline app/services/workflow_executions/preparation_service.rb:26]: Due to now needing an additional rubocop disable, I think we can just move checking `@pipeline.nil?`  into its own method.

Can you replace this with `return false unless validate_pipeline`

And then under `private`:
```
def validate_pipeline
      return true unless @pipeline.nil?

      @workflow_execution.state = :error
      @workflow_execution.cleaned = true
      @workflow_execution.save
      false
    end
```

**Review by ChrisHuynh333 — CHANGES_REQUESTED** [review]
Works great! Just one last refactor from me

**Review by ChrisHuynh333 — APPROVED** [review]
Looks good!

File: config/pipelines/pipelines.json [hunk 2]
```diff
@@ -4,6 +4,10 @@
     "name": "phac-nml/iridanextexample",
     "description": "IRIDA Next Example Pipeline",
     "versions": [
+      {
```
Comment (deepsidhu85) [inline config/pipelines/pipelines.json:7]: Can we remove the `"executable": false` from this pipeline and change one of the earlier versions to include it instead? For testing purposes we probably want the newest version to be executable

File: app/services/workflow_executions/preparation_service.rb [hunk 3]
```diff
@@ -52,10 +57,18 @@ def execute # rubocop:disable Metrics/MethodLength
...
-    def parse_samplesheet_headers
-      workflow = Irida::Pipelines.instance.find_pipeline_by(@workflow_execution.metadata['workflow_name'],
-                                                            @workflow_execution.metadata['workflow_version'])
-      workflow.samplesheet_headers
+    def find_pipeline
+      Irida::Pipelines.instance.find_pipeline_by(@workflow_execution.metadata['workflow_name'],
+                                                 @workflow_execution.metadata['workflow_version'])
+    end
+
+    def validate_pipeline
```
Comment (deepsidhu85) [inline app/services/workflow_executions/preparation_service.rb:65]: We probably want to update the job to also set the state to error as the pipeline can be marked as non-executable after submission and before the preparation job actually runs

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking good just a few comments below

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me!

**Pull request merged**